docs/proposals/enable-local-source-code-support.md

```diff
@@ -0,0 +1,302 @@
+
+ 2. A `spec.source.container.path` mandatory [field](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/apis/build/v1alpha1/source.go#L58) for specifying where the source code is located in the local workstation.
+
+ 3. A `spec.source.container.image` mandatory [field](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/apis/build/v1alpha1/source.go#L49) for specifying the container image endpoint, registry and repository to build. If an image is pinned to a _SHA_, we could assume the image already exists, and we will only _pull_ it, and not longer _push_ it.
+
+ On top of that, to fulfill the features defined above, we could extend the API to include:
+
+ 4. A `spec.source.container.skipDirs` optional field. To provide a list of directories that we shouldn´t bundle.
+
+ 5. A `spec.source.container.pruneBundle` optional field. To signalize the deletion of the bundle image at the very end of the execution of a strategy. Should be disable by _default_.
+
+> Make `spec.source.url` a none mandatory field
+
+ 1. This was done thinking on assets only hosted in `git`, which no longer holds true. See an [example](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/apis/build/v1alpha1/source.go#L19-L20).
+
+### Proposal 02: Runtime Modifications
+
+> Ensure that the bundle image is prepended as the first step in the Build strategies.
+
+  1. When `spec.source.container.path` is defined, we should  not longer create the Tekton Input PipelineResource. We do this today, to tell Tekton that we want to pull source from a git repository, which ends as a container that pulls it. See an [example](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/reconciler/buildrun/resources/taskrun.go#L184-L185) on future changes.
+  
+  2. We need to **prepend** a new step in our Task step definition, which will pull our local source code from the bundle image. See an [example](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/reconciler/buildrun/resources/taskrun.go#L171-L186). Important to notice, the image to pull will be a self-extracted image, therefore the `workingDir` container definition should be under `/workspace/source`, which is a well-known path in the Shipwright strategies, where source code is expected to be.
+
+  3. If authentication was specified for the bundle image, we need to ensure we have that mounted in the pod either in the form of a secret or a service-account.
+
+### Proposal 03: Self-Extractable Base Image
+
+> Build an image that will serve as the base layer for a bundle image
+
+  1. Build an image that can self-extract code, by walking the existing tarball and copying the contents into a particular directory.
```

Comment:
The *particular directory*  will likely be `$(params.shp-source-root)` = /workspace/source. Although we do not plan to get away from this directory, I think it makes sense if the self-extract code does not have this hard-coded but takes this as an argument.